code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -1324,3 +1324,232 @@
 	var/mob/living/carbon/xenomorph/X = owner
 	X.hive.purchases.interact(X)
 	return succeed_activate()
+
+// ***************************************
+// *********** Tail Stab
+// ***************************************
+//totally not stolen from punch code
+/datum/action/ability/activable/xeno/tail_stab
+	name = "Tail Stab"
+	action_icon_state = "tail_attack"
+	desc = "Strike a target within two tiles with a sharp tail for armor-piercing damage, stagger and slowdown. Deals double AP, damage, stagger and slowdown to grappled targets, structures and machinery."
+	ability_cost = 30
+	cooldown_duration = 10 SECONDS
+	use_state_flags = ABILITY_USE_AGILITY
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_TAIL_STAB,
+	)
+	target_flags = ABILITY_MOB_TARGET|ABILITY_MOB_TARGET
+	var/range = 2
+	var/punch_description = "swift tail-jab!"
```

Comment:
autodoc. also, punch_description for a stab?

Reply:
weird i had changed it to stab desc, probably github magic from commit things. I'll re change lmao

Reply:
must be from when i tried to make this a punch alt and forgot to clean that up back to stab when it didnt work so well